packages/ui/src/components/label-marker.tsx

```diff
@@ -25,23 +29,73 @@ export const LabelMarkerColor = {
   [ColorsEnum.MINT]: 'bg-label-background-mint text-label-foreground-mint',
   [ColorsEnum.LIME]: 'bg-label-background-lime text-label-foreground-lime'
 }
+const deleteButtonColor = {
+  [ColorsEnum.RED]: 'text-label-foreground-red hover:opacity-60',
+  [ColorsEnum.BLUE]: 'text-label-foreground-blue hover:opacity-60',
+  [ColorsEnum.GREEN]: 'text-label-foreground-green hover:opacity-60',
+  [ColorsEnum.YELLOW]: 'text-label-foreground-yellow hover:opacity-60',
+  [ColorsEnum.PURPLE]: 'text-label-foreground-purple hover:opacity-60',
+  [ColorsEnum.PINK]: 'text-label-foreground-pink hover:opacity-60',
+  [ColorsEnum.VIOLET]: 'text-label-foreground-violet hover:opacity-60',
+  [ColorsEnum.INDIGO]: 'text-label-foreground-indigo hover:opacity-60',
+  [ColorsEnum.CYAN]: 'text-label-foreground-cyan hover:opacity-60',
+  [ColorsEnum.ORANGE]: 'text-label-foreground-orange hover:opacity-60',
+  [ColorsEnum.BROWN]: 'text-label-foreground-brown hover:opacity-60',
+  [ColorsEnum.MINT]: 'text-label-foreground-mint hover:opacity-60',
+  [ColorsEnum.LIME]: 'text-label-foreground-lime hover:opacity-60'
+}
+
+export const LabelMarker: FC<LabelMarkerProps> = ({
+  type = LabelType.STATIC,
+  label,
+  value,
+  color,
+  counter,
+  onDelete,
+  className
+}) => {
+  const isDynamic = type === LabelType.DYNAMIC
+  const isWithExtraContent = !!value || !!counter
+  const isWithDeleteButton = !!onDelete
 
-export const LabelMarker: FC<LabelMarkerProps> = ({ label, value, color, inTable = false }) => {
   return (
-    <span
-      className={cn(
-        `inline-flex text-12 leading-5 h-5 py-px px-2 items-center font-medium rounded overflow-hidden max-w-full ${LabelMarkerColor[color]}`,
-        { 'pr-px': !!value }
-      )}
-    >
-      <span className={cn('truncate', { 'flex-1': inTable }, { 'pr-1.5': !!value })}>{label}</span>
-      {!!value && (
-        <span
-          className={`flex h-full items-center overflow-hidden rounded-r bg-label-background-black pl-1.5 pr-[7px] ${inTable && 'max-w-[50%]'}`}
+    <div className={cn('flex max-w-full items-center gap-2 opa relative', className)}>
+      <span
```

Comment:
This being a `span` (an inline element), but being set as display grid (a block level element) doesn't really make much sense to me. Is there a reason we can't use a `div`? Makes no difference, but I'm curious if there's a reason as I've seen inline elements being changed to block a lot in this codebase 🤔